src/main/java/org/graylog/integrations/pagerduty/PagerDutyNotification.java

```diff
@@ -0,0 +1,106 @@
+import org.graylog.events.notifications.PermanentEventNotificationException;
+import org.graylog.events.notifications.TemporaryEventNotificationException;
+import org.graylog.integrations.pagerduty.client.MessageFactory;
+import org.graylog.integrations.pagerduty.client.PagerDutyClient;
+import org.graylog.integrations.pagerduty.dto.PagerDutyResponse;
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.system.NodeId;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Main class that focuses on event notifications that should be sent to PagerDuty.
+ *
+ * @author Edgar Molina
+ *
+ */
+public class PagerDutyNotification implements EventNotification
+{
+    private final PagerDutyClient pagerDutyClient;
+    private final MessageFactory messageFactory;
+    private final ObjectMapper objectMapper;
+    private final NotificationService notificationService;
+    private final NodeId nodeId;
+
+    @Inject
+    PagerDutyNotification(PagerDutyClient pagerDutyClient,
+                          MessageFactory messageFactory,
+                          ObjectMapper objectMapper,
+                          NotificationService notificationService,
+                          NodeId nodeId) {
+        this.pagerDutyClient = pagerDutyClient;
+        this.messageFactory = messageFactory;
+        this.objectMapper = objectMapper;
+        this.notificationService = notificationService;
+        this.nodeId = nodeId;
+    }
+
+    public interface Factory extends EventNotification.Factory {
+        @Override
+        PagerDutyNotification create();
+    }
+
+    @Override
+    public void execute(EventNotificationContext ctx) throws EventNotificationException {
+        final String payloadString = buildRequestBody(ctx);
+        try {
+            PagerDutyResponse response = pagerDutyClient.enqueue(payloadString);
+            List<String> errors = response.getErrors();
+            if (errors != null && errors.size() > 0) {
+                throw new IllegalStateException(
+                        "There was an error triggering the PagerDuty event, details: " + errors);
+            }
+        } catch (PagerDutyClient.TemporaryPagerDutyClientException e) {
+            throw new TemporaryEventNotificationException(
+                    String.format("Error enqueueing the PagerDuty event :: %s", e.getMessage()),
+                    null != e.getCause() ? e.getCause() : e);
+        } catch (PagerDutyClient.PermanentPagerDutyClientException e) {
+            String errorMessage = String.format("Error enqueueing the PagerDuty event :: %s", e.getMessage());
+            Notification systemNotification = notificationService.buildNow()
+                    .addNode(nodeId.toString())
+                    .addType(Notification.Type.GENERIC)
+                    .addSeverity(Notification.Severity.URGENT)
```

Comment:
how do we determine the severity is `urgent` or `normal`?

Reply:
Generally speaking, if you are sending something to PagerDuty it is because you want your ops people to be aware of it immediately.  Therefore, if you have a problem that will prevent anything from being sent to PagerDuty, that is an urgent issue that requires immediate attention.